File: app/src/main/java/es/headfon/headfones/SearchSpotifyActivity.java
package es.headfon.headfones;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.KeyEvent;
import android.view.View;
import android.widget.EditText;

import org.json.JSONObject;

public class SearchSpotifyActivity extends AppCompatActivity {
    private String access_token;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_search_spotify);
        Intent intent = getIntent();
        access_token = intent.getExtras().getString("access_token");
    }

    @Override
    public boolean dispatchKeyEvent(KeyEvent e) {
        if (e.getKeyCode() == KeyEvent.KEYCODE_ENTER) {
            onSearchClick(findViewById(R.id.searchText));
            return true;
        }
        return super.dispatchKeyEvent(e);
    };

    public void onSearchClick(View view) {
        view.clearFocus();
        EditText search  = findViewById(R.id.searchText);
        String url = "https://api.spotify.com/v1/search/?q=" + search.getText().toString() + "*&type=album,artist,playlist,track";
        HttpUtility http = new HttpUtility();

        http.makeRequest(url, access_token, new HttpUtility.DataCallbackInterface() {
            @Override
            public void onDataFetch(JSONObject result) {
                Intent intent = new Intent(getApplicationContext(), SearchResultsActivity.class);
                intent.putExtra("search_results", result.toString());
                startActivity(intent);
            }
        });

    }
}

File: app/src/main/java/es/headfon/headfones/SearchSpotifyActivity.kt
package es.headfon.headfones

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.KeyEvent
import android.view.View
import android.widget.EditText

import org.json.JSONObject

class SearchSpotifyActivity : AppCompatActivity() {
    private var access_token: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_search_spotify)
        val intent = intent
        access_token = intent.extras!!.getString("access_token")
    }

    override fun dispatchKeyEvent(e: KeyEvent): Boolean {
        if (e.keyCode == KeyEvent.KEYCODE_ENTER) {
            onSearchClick(findViewById(R.id.searchText))
            return true
        }
        return super.dispatchKeyEvent(e)
    }

    fun onSearchClick(view: View) {
        view.clearFocus()
        val search = findViewById<EditText>(R.id.searchText)
        val url = "https://api.spotify.com/v1/search/?q=" + search.text.toString() + "*&type=album,artist,playlist,track"
        val http = HttpUtility()

        http.makeRequest(url, access_token!!, object : HttpUtility.DataCallbackInterface {
            override fun onDataFetch(result: JSONObject) {
                val intent = Intent(applicationContext, SearchResultsActivity::class.java)
                intent.putExtra("search_results", result.toString())
                startActivity(intent)
            }
        })

    }
}
